app/models/admin.rb

```diff
@@ -1,18 +1,42 @@
 class Admin < ActiveRecord::Base
   include ActiveModel::ForbiddenAttributesProtection
-  # Authlogic gem
-  acts_as_authentic do |config|
-    config.transition_from_restful_authentication = true
-    if (ArchiveConfig.BCRYPT  || "true")  == "true" then
-      config.crypto_provider = Authlogic::CryptoProviders::BCrypt
-      config.transition_from_crypto_providers = [Authlogic::CryptoProviders::Sha512, Authlogic::CryptoProviders::Sha1]
-    else
-      config.crypto_provider = Authlogic::CryptoProviders::Sha512
-      config.transition_from_crypto_providers = [Authlogic::CryptoProviders::Sha1]
+
+  devise :database_authenticatable,
+         :registerable,
+         :validatable,
+         password_length: ArchiveConfig.PASSWORD_LENGTH_MIN..ArchiveConfig.PASSWORD_LENGTH_MAX
+
+  # http://stackoverflow.com/questions/6113375/converting-existing-password-hash-to-devise
+  # https://github.com/binarylogic/authlogic/blob/master/lib/authlogic/acts_as_authentic/password.rb#L361
+  # https://www.ruby-forum.com/topic/217465
+  # Not useful but https://github.com/plataformatec/devise/issues/511
+
+  alias :devise_valid_password? :valid_password?
+  def valid_password?(password)
+    begin
+      result = super(password)
+      # Now the common form is that we are using an authlogic method so lets
+      # test that on failure
+      return true if result
+      if Authlogic::CryptoProviders::BCrypt.matches?(self.encrypted_password,[password, self.password_salt].compact)
```

Comment:
I don't know enough about encryption to judge if this weakens our security or not. Your comment implies that you did your research and I trust you on that.